### DISCUSSION TITLE
Is there a better way to disable HTTP keep-alive? (connection: close)

### DISCUSSION THREAD

**Discussion by nwtgck** [nodejs/undici#1422] [Q&A]
Hi, `undici.request()` requests with `connection: keep-alive` by default. Specifying `headers: { 'connection': 'close' }` causes "InvalidArgumentError: invalid connection header" error.

I found `dispatcher: new undici.Agent({pipelining: 0})`disables keep-alive as follows.

```js
const undici = require("undici");

(async () => {
  const res = await undici.request("http://localhost:8181", {
    dispatcher: new undici.Agent({pipelining: 0}),
    body: "hello",
  });
})();
```

```console
$ nc -lp 8181
PUT / HTTP/1.1
host: localhost:8181
connection: close
content-length: 5

hello
```

Is there any better way to disable HTTP keep-alive? I mean "better" is that it is clear to understand for everyone, `new Agent({pipelining: 0})` is not clear for me.

**Comment by iitalics** [comment]
recently ran into this, i think you can use `headers: { connection: 'close' }` or `reset: true` as of https://github.com/nodejs/undici/pull/1829
